### PR TITLE
Update core and object store and merge master

### PR DESCRIPTION
**Pull request by tgoyne**
This turned out to be rather involved as there were several breaking changes in OS that we hadn't pulled in yet.

### CODE REVIEW

File: Realm/RLMMongoDatabase_Private.hpp [hunk 1]
```diff
@@ -17,18 +17,17 @@
...
+@class RLMUser;
 
 @interface RLMMongoDatabase ()
 
-@property (nonatomic, strong) RLMApp *app;
+@property (nonatomic, strong) RLMUser *user;
 @property (nonatomic, strong) NSString *serviceName;
 
-- (instancetype)initWithApp:(RLMApp *)app
+- (instancetype)initWithUser:(RLMUser *)user
```
Comment (leemaguire) [inline Realm/RLMMongoDatabase_Private.hpp:30]: indentation looks slightly off

**Review by leemaguire — APPROVED** [review]

File: Realm/ObjectServerTests/RLMObjectServerTests.mm [hunk 2]
```diff
@@ -515,7 +513,7 @@ - (void)testInvalidPasswordAuthentication {
 
     XCTestExpectation *expectation = [self expectationWithDescription:@"login should fail"];
 
-    [self.app loginWithCredential:credentials completion:^(RLMUser * user, NSError * error) {
+    [self.app loginWithCredential:credentials completion:^(RLMUser * user, NSError *error) {
```
Comment (jsflax) [inline Realm/ObjectServerTests/RLMObjectServerTests.mm:516]: There are a bunch of extra spaces in this file before `*`, but they were already there

File: Realm/RLMMongoCollection.mm [hunk 3]
```diff
@@ -93,31 +95,41 @@ - (void)close {
 
 @implementation RLMMongoCollection
 
-- (instancetype)initWithApp:(RLMApp *)app
-                serviceName:(NSString *)serviceName
-               databaseName:(NSString *)databaseName
-             collectionName:(NSString *)collectionName {
+static realm::bson::BsonDocument toBsonDocument(id<RLMBSON> bson) {
```
Comment (jsflax) [inline Realm/RLMMongoCollection.mm:98]: Should this be prefixed by `RLM`?

File: RealmSwift/RealmConfiguration.swift [hunk 4]
```diff
@@ -200,7 +200,11 @@ extension Realm {
         /// The classes managed by the Realm.
         public var objectTypes: [ObjectBase.Type]? {
             get {
-                return self.customSchema.map { $0.objectSchema.compactMap { $0.objectClass as? Object.Type } }
+                return self.customSchema.map {
+                    $0.objectSchema.compactMap {
+                        $0.objectClass as? Object.Type ?? $0.objectClass as? EmbeddedObject.Type
```
Comment (jsflax) [inline RealmSwift/RealmConfiguration.swift:205]: Can we just leave a comment here saying what this is doing? It's fairly obvious when reading it with context, but it certainly looks strange.

File: Realm/RLMMongoCollection.mm (hunk 3 above)
Comment (tgoyne) [reply Realm/RLMMongoCollection.mm:98]: It's static, and only exported functions should be prefixed.

**Pull request merged**